### PR TITLE
ocamlPackages: add meta.mainProgram to many packages

### PR DESCRIPTION
**Pull request by malob**
###### Description of changes

Continuing on my adventure to get `nix run` working with more packages in `nixpkgs` (previously #171602), I've run my hacky script on `ocamlPackages`.

This PR adds `meta.mainProgram` to all packages in `ocamlPackages` available for Darwin that provide a single binary whose name differs from the package's `name` or `pname`.

(While I was at it, I also ended up tweaking the order of the meta attributes to my understanding of which order is best practice.)

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by malob** [comment]
Result of `nixpkgs-review pr 172228` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/development/ocaml-modules/uuidm/default.nix [hunk 1]
```diff
@@ -18,7 +18,8 @@ stdenv.mkDerivation rec {
     description = "An OCaml module implementing 128 bits universally unique identifiers version 3, 5 (name based with MD5, SHA-1 hashing) and 4 (random based) according to RFC 4122";
     homepage = "https://erratique.ch/software/uuidm";
     license = licenses.bsd3;
-    platforms = ocaml.meta.platforms or [];
     maintainers = [ maintainers.maurer ];
+    mainProgram = "uuidtrip";
+    platforms = ocaml.meta.platforms or [];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/uuidm/default.nix:23]:
```suggestion
    platforms = ocaml.meta.platforms;
```

File: pkgs/development/ocaml-modules/xmlm/default.nix [hunk 2]
```diff
@@ -27,8 +27,9 @@ stdenv.mkDerivation rec {
   meta = with lib; {
     description = "An OCaml streaming codec to decode and encode the XML data format";
     homepage = webpage;
-    platforms = ocaml.meta.platforms or [];
-    maintainers = [ maintainers.vbgl ];
     license = licenses.bsd3;
+    maintainers = [ maintainers.vbgl ];
+    mainProgram = "xmltrip";
+    platforms = ocaml.meta.platforms or [];
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/xmlm/default.nix:33]:
```suggestion
    platforms = ocaml.meta.platforms;
```

**Review by vbgl — APPROVED** [review]
AFAICT, this makes no harm.

File: pkgs/development/ocaml-modules/uuidm/default.nix (hunk 1 above)
Comment (malob) [reply pkgs/development/ocaml-modules/uuidm/default.nix]: I noticed that there were many more cases of this in `ocamlPackages` so I've gone ahead and fixed them all by changing them to `inherit (ocaml.meta) platforms;` in a new commit.

**Pull request merged**